packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -413,6 +414,16 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                                     const age = Date.now() - (snapshotCachedEntry.cacheEntryTime ??
                                         (Date.now() - 30 * 24 * 60 * 60 * 1000));
 
+                                    // In order to decrease the number of times we have to execute a snapshot refresh,
+                                    // if this is the summarizer and we have a cache entry but it is past the defaultFirstSummaryCacheExpiryTimeout,
+                                    // force the network retrieval instead as there might be a more recent snapshot available.
+                                    if (this.hostPolicy.summarizerClient &&
```

Comment:
@vladsud Can you check / confirm that updating the code here is enough to fix this issue?

Reply:
It looks fine to me. That said, some feedback:
1. It's useful to put issue references in the code, as they have way more data about reasoning, stats, etc. Something that comes handy next time when we need to change it or question the decision
2. It would be great to have telemetry when we reject it, to be able to assess if we should tune the number upwards.